### DISCUSSION TITLE
Bunder header version compatibility fails issue

### DISCUSSION THREAD

**Discussion by kevison** [dotnet/runtime#85789] [General]
We have a situation where applications that are published with "Produce single file"  fail when a user tries to install on a machine that just had the .net6 binaries (runtime/desktop/bundle(x86 and x64)) installed.  Could this be related to issue #70039? 

We are developing using VS2022, .NET6.  Target winx86.

**Comment by jkotas** [comment]
Any error messages? 

> Could this be related to issue https://github.com/dotnet/runtime/issues/70039?

Unlikely.

**Comment by kevison** [comment]
As to error messages... from the event logs:

  Description: A .NET application failed. Application: EnhancedInstallUtil.exe Path: C:\LPS\LPS_Interfaces\LSS\Host\EnhancedInstallUtil.exe Message: Failure processing application bundle. Bundle header version compatibility check failed. A fatal error occured while processing application bundle</Data>


Also I refer to this article on stack overflow [Bundle header issue](https://stackoverflow.com/questions/76167641/how-to-resolve-bundle-header-version-compatibility-check-failed)  and this [issue](https://github.com/dotnet/runtime/issues/59819)

**Reply by vitek-karas** [reply]
Do you use framework dependent or self-contained single-file? (So you set SelfContained=false for example in your project, another simple way to tell is by the size of the exe - self-contained single-file will typically be pretty large, like 60MB or larger, unless you also use trimming).

Does the machine have .NET 6 installed for the right architecture?

If you can repro this, could you please share a host trace? https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#corehost_trace. Please note that the trace may contain information about your machine and user (like environment variables, paths, and so on), make sure you're OK with that before sharing it.
